processor/transaction/span_context.go

```diff
@@ -0,0 +1,30 @@
+package transaction
+
+import (
+	m "github.com/elastic/apm-server/model"
+	"github.com/elastic/apm-server/utility"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type SpanContext struct {
+	service common.MapStr
+}
+
+func NewSpanContext(service *m.Service) *SpanContext {
+	return &SpanContext{service: service.MinimalTransform()}
+}
+
+func (c *SpanContext) Transform(m common.MapStr) common.MapStr {
+	if m == nil {
+		m = common.MapStr{}
+	} else {
+		for k, v := range m {
+			utility.Add(m, k, v)
+		}
+	}
+	utility.Add(m, "service", c.service)
+	if len(m) == 0 {
```

Comment:
same as before, this should be needed

Reply:
It is necessary as it checks whether ot not `m` has elements, if this check is removed an empty hash would be returned instead of `nil` and stored in ES.

Reply:
why? aren't you already doing `if len(valMap) > 0 `  in `utility.Add`?

Reply:
`m` is the base map to which everything is added. If nothing is added `nil` should be returned.

Reply:
ok, i don't get it. i commented out all the `if len(m) == 0` in this PR, and all the approval tests pass. Can you add an approval test that exercise this code path? I don't understand how affects the final result.

Reply:
you're right that this is checked when the `context` is added to the events, removed it.